stingray/crossspectrum.py

```diff
@@ -500,13 +617,23 @@ def _make_crossspectrum(self, lc1, lc2, fullspec=False):
 
         # total number of photons is the sum of the
         # counts in the light curve
+        self.meancounts1 = lc1.meancounts
+        self.meancounts2 = lc2.meancounts
         self.nphots1 = np.float64(np.sum(lc1.counts))
         self.nphots2 = np.float64(np.sum(lc2.counts))
 
-        self.meancounts1 = lc1.meancounts
-        self.meancounts2 = lc2.meancounts
+        self.err_dist = 'poisson'
+        if lc1.err_dist == 'poisson':
+            self.var1 = lc1.meancounts
+        else:
+            self.var1 = np.mean(lc1.counts_err) ** 2
+            self.err_dist = 'gauss'
```

Comment:
In lightcurve we do define err_dist to poisson when not specified, so this should always work.
I used this structure because I assume anything that is not Poisson is "more similar to a Gaussian" (i.e. being able to specify a variance/standard deviation) than to Poisson. I know it's a bold assumption, but I assume that if they give an array of counts error, they know what they are doing (another bold assumption :D).
Does it sound right? No strong opinions.

Reply:
Yes, I agree that this in principle works, and will always return a valid result, but I think it's still a logical jump where I'd as a user expect consistent behaviour. We should decide whether stingray assumes Poisson except when the user explicitly specifies something else, or Gaussian unless the user explicitly specifies something else, but I'd rather not have two different behaviours across two classes?

Reply:
However, it is very difficult to encode this here. The current version says: Light curve is Poisson? Then var is equal to mean. Otherwise, use error bars. There are no assumptions involved. If we make assumptions here, I'm not sure what could go wrong, but something sooner or later would :D. Also, are you sure the user would see any logical jump here? It's mostly transparent to them, and decided by how they defined the light curve (and if they didn't then we decided for them that it was Poisson).

Reply:
(In the meantime, I rebased)